### DISCUSSION TITLE
Publish an app route as the root of a website

### DISCUSSION THREAD

**Discussion by laem** [vercel/next.js#60358] [App Router]
Hi, I'm trying to do something fairly simple but haven't been able to find a solution on forums. 

I have a main Next repo, with multiple routes. One of them is `app/studio` with its page.tsx and subroutes. 

I was to deploy this `studio` route as the root of a new domain, let's say `awesome.studio`. Hitting `/` on awesome.studio would serve what's in the monorepo's `app/studio/` root. 

Context : it's kind of the new version of my website. I want to be able to develop it in the main repo, but publish it on a new domain that will eventually replace the first. 

Thanks for your help.

**Comment by JCharante** [comment]
I think this document page about creating rewrites in next.config.js could solve your problem, if not then let me know https://nextjs.org/docs/app/api-reference/next-config-js/rewrites

**Comment by laem** [comment]
Hi, thanks for your help, `rewrites` are indeed the good tool. 

However, I can't make it work to redirect from the root page. 

```
	async rewrites() {
		return [
			{
				source: '/:path*',
				destination: '/studio/:path*',
			},
		]
	},

```

Hitting `mydomain.org/test` will show `mydomain.org/studio/test`, but hitting the root `mydomain.org` won't show `mydomain.org/studio`. It just show the `app/page.tsx` main page. 

I've also tried with just `source: '/'`, didn't work either. I'm using Next `14.0.4`.

**Reply by laem** [reply]
OK, got it ! 

I needed to use ` beforeFiles`, since my `app/page.tsx` is a static page getting served before redirects, if I understand well. 

The doc could use a clearer mention of this, I guess ?
